typescript/fastrpc.ts

```diff
@@ -327,8 +343,18 @@ function _serializeArrayBuffer(result: BYTES, data: ArrayBuffer) {
 	return;
 }
 
-function _serializeArray(result: BYTES, data: any[]) {
-	if (_getHint() == "binary") {
+function _serializeArray(result: BYTES, data: any[] | Binary) {
```

Comment:
Dva komentare:
  - neni implementace pro varianty `isBinary` a `_getHint() == "binary"`  stejna? Pokud ano, nemusela by byt ve dvou ruznych vetvich...
  - neni nahodou ten vypocet prvnich par bajtu spolecny pro vsechny tri vetve? Pokud ano, dal by se udelat spolecne (a lisila by se pak jen serializace samotneho pole, bud pres `_append` cele promenne, nebo cyklem pro `ArrayBuffer`...

Reply:
Podivej se jak jsem to zkusil vyřešit

Reply:
Nekam tam uplne zmizelo to `TYPE_ARRAY << 3;`, ne?

Asi ta funkce `_serializeArray` zacina byt prilis chytra: na vstupu prijima bud JS pole hodnot, nebo JS pole s hintem "binary", nebo JS objekt s `_hint:"binary"`. A na vystupu generuje bud TYPE_ARRAY nebo TYPE_BINARY.

Co kdybychom to teda rozdelili na dve? Jedna, ktera *vzdy* produkuje TYPE_ARRAY, a druha, ktera *vzdy* produkuje TYPE_BINARY (a do ni se posilaji arraybuffery a obe dve hintovane varianty).

Reply:
Jo to dává smysl protože i tak ta funkce `_serializeArray` byla spíše sloučeninou dvou funkcí. takhle to bude přehlednější.